### PR TITLE
Fix streaming propagate + fixing bug in KV Cache

### PR DESCRIPTION
**Pull request by adefossez**
## Checklist

- [x] Read CONTRIBUTING.md, and accept the CLA by including the provided snippet. We will not accept PR without this.
- [x] Run pre-commit hook.
- [ ] If you changed Rust code, run `cargo check`, `cargo clippy`, `cargo test`.

## PR Description

Depformer was never set in streaming mode which explain deterioration in quality in the CUDA demo (weights from first cb used all the way + no kv cache on previous cb...). Sorry about that, I'm quite surprised how it still somehow worked not so bad despite that pretty massive change.
Also fixing a bug in the KVCache, not as bad, would only have impact when wrapping around based on unit tests I ran on the internal codebase.

Update: after further review, the bug had no impact except if sin embeddings were to be used in the depformer which is not the case. This is because the original function was a mess and still properly put any children of Depformer in streaming mode even if Depformer itself was not streaming

### CODE REVIEW

File: moshi/moshi/__init__.py [hunk 1]
```diff
@@ -15,4 +15,4 @@
 from . import models
 from . import quantization
 
-__version__ = "0.1.0"
+__version__ = "0.1.1a1"
```
Comment (LaurentMazare) [inline moshi/moshi/__init__.py:18]: Why not `0.1.2`?

File: moshi/moshi/modules/streaming.py [hunk 2]
```diff
@@ -64,25 +64,26 @@ def is_streaming(self):
...
     def _apply_named_streaming(self, fn: tp.Any):
-        def _handle_module(prefix: str, module: nn.Module, recurse: bool = True):
-            propagate = True
+        def _handle_module(prefix: str, module: nn.Module):
             if isinstance(module, StreamingModule):
-                if module._streaming_propagate:
-                    fn(prefix, module)
+                # If prefix is empty, we are the direct receiver of the streaming request,
+                # otherwise, we are inheriting from a parent and will stop if _streaming_propagate is False.
+                if not module._streaming_propagate and prefix != "":
```
Comment (LaurentMazare) [inline moshi/moshi/modules/streaming.py:76]: I find the semantics pretty confusing here: if A has propagate set, and has a children B for which propagate is not set, then calling `A._apply_named_streaming` results in `fn(A)` but not in `fn(B)`. I would have expected `fn(B)` to be called (but not `fn(C)` for any child C of B as B doesn't propagate down).
Anyway what is the use case for not propagating? Shouldn't we just always propagate?

Comment (adefossez) [reply moshi/moshi/modules/streaming.py:76]: maybe a better name could be `streaming_detach`, closer to detaching a tensor from a compute graph for the backward. It means this entire module should not be consider a part of the main streaming path, in our case because the depformer is not streaming along the time dim, but some other dim

File: moshi/moshi/__init__.py (hunk 1 above)
Comment (LaurentMazare) [reply moshi/moshi/__init__.py:18]: Gotcha, sounds like a reasonable naming scheme so let's go ahead with this.

File: moshi/moshi/modules/streaming.py (hunk 2 above)
Comment (LaurentMazare) [reply moshi/moshi/modules/streaming.py:76]: +1 for `streaming_detach`

Comment (adefossez) [reply moshi/moshi/modules/streaming.py]: done

**Review by LaurentMazare — APPROVED** [review]
Nice catch, thanks!
